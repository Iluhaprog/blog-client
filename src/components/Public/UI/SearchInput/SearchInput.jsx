import React from 'react';
import styled from 'styled-components';
import {
  ReactComponent as Search,
} from '../../../../assets/public/icons/search.svg';
import PropTypes from 'prop-types';

const SearchInputBox = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  max-width: ${({maxWidth}) => maxWidth + 'px' || 'fit-content'};
  width: 100%;
  align-items: center;
  border: 2px solid var(--color-1);
  box-sizing: border-box;
  border-radius: ${({mod}) => {
    return mod === 'big' ? 15 : 5;
  }}px;
  padding: ${({mod}) => {
    return mod === 'big' ? 10 : 5;
  }}px;
  
  input {
    outline: none;
    width: 100%;
    margin: 0;
    border: 0;
    font-size: ${({mod}) => {
    return mod === 'big' ? 16 : 12;
  }}px;
    
    &::placeholder {
      color: rgba(0, 0, 0, .4);
    }
  }
  
  button {
    border: 0;
    margin: 0;
    padding: 0;
    background-color: transparent;
  }
`;

const SearchInput = ({
  name,
  value,
  onChange,
  placeholder,
  mod,
  maxWidth,
}) => {
  return (
    <SearchInputBox maxWidth={maxWidth} mod={mod}>
      <input
        name={name}
        value={value}
        onChange={onChange}
        placeholder={placeholder}
      />
      <button type={'submit'}>
        <Search />
      </button>
    </SearchInputBox>
  );
};

SearchInput.propTypes = {
  mod: PropTypes.oneOf(['big', 'small']),
  name: PropTypes.string,
  value: PropTypes.string,
  onChange: PropTypes.func,
  placeholder: PropTypes.string,
  maxWidth: PropTypes.number,
};

export {SearchInput};
