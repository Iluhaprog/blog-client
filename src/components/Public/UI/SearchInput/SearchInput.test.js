import {render} from '@testing-library/react';
import {SearchInput} from './SearchInput';
import React from 'react';

test('Should render SearchInput component', () => {
  const {asFragment} = render(
    <SearchInput
      maxWidth={500}
      value={''}
      placeholder={'Search'}
      onChange={() => alert('c')}
      mod={'big'}
    />
  );
  expect(asFragment()).toMatchSnapshot();
});
