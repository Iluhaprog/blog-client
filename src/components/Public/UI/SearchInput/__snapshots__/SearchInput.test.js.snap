// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render SearchInput component 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ eUnQoZ"
  >
    <input
      placeholder="Search"
      value=""
    />
    <button
      type="submit"
    >
      <svg>
        search.svg
      </svg>
    </button>
  </div>
</DocumentFragment>
`;
